chainer/functions/connection/convolution_2d.py

```diff
@@ -33,7 +33,7 @@ def _pair(x):
 
 class Convolution2DFunction(function.Function):
 
-    def __init__(self, stride=1, pad=0, cover_all=False, **kwargs):
+    def __init__(self, stride=1, pad=0, cover_all=False, dilate=1, **kwargs):
```

Comment:
It seems you put `dilate` option in `Convolution2DFunction`, but it is not supported in CPU mode. This is misleading for users. Further, if `Convolution2DFunction` supports dilation feature fully, then `DilatedConvolution2DFunction` will become the subset of it. This means we will have some sort of overlapping codes. I think there are two ways to fix this.

1. Do not modify `Convolution2DFunction` (i.e., do not put `dilate` option here) and modify only the dilated function.
2. Complete to implement `dilate` argument (i.e., implement dilation in CPU mode) and make the current dilated function just calls `Convolution2DFunction` internally.

How do you think about this?